### DISCUSSION TITLE
Tab coloring in VSCode Extension

### DISCUSSION THREAD

**Discussion by XXDIL** [microsoft/vscode-discussions#312] [Extension Development QnA] [answered]
I wanted to get some insight on how I could color multiple tabs in vscode with different colors.

**More specifically:**
1) Each tab is associated with a specific connection.
2) A specific connection belongs to a specific color.
3) So when I open a tab that has a connection that's mapped to green (for example), that tab should be in green.

I have already looked into an approach that has to do with updating the `settings.json` file, and using `tab.activeBackground`.
I was wondering if there is any other way to do this programatically, an approach that does'nt seem so forced.

Basically I don't want to modify the settings.json file repeatedly. (Or ideally not edit it at all)

**Comment by alefragnani** [comment, marked as answer]
Hi @XXDIL ,

The current Tabs API doesn't allow you to change its color so, for now, you will be forced to touch `settings.json` in order to define custom colors. But, it only supports/exposes two tab states (active, inactive), and the _group_ concept it has today, if I remember correctly, is related to the four groups where you can split/organize editors. I'm not sure this could be updated/replaced for this _group of tabs_ you need.

That being said, I would suggest you to open an issue in VS Code repo, asking for a new/updated API

Hope this helps

**Comment by r3m0t** [comment]
Did you look at the file decoration API which lets you color the filename, which is in the Explorer and the foreground of the tabs?

**Reply by XXDIL** [reply]
I have looked into it but that was for coloring the tree nodes, could you provide a sample of how I can do that for tab color?

**Reply by r3m0t** [reply]
No it's not possible, I was just suggesting an alternative

**Comment by mondersky** [comment]
If you want to color your vs code tabs you can't do that with the current API you have to do some patching, I made the same in my extension https://github.com/mondersky/tabscolor-vscode
